Sorting_Algo_withTimings/SortTiming.java

```diff
@@ -0,0 +1,136 @@
+ * 
+ * @author Ibrahim Hossain Limon
+ * @version 23 Aug 2023
+ */
+public class SortTiming {
+    /** Supported algorithms */
+    public enum Alg { BUBBLE, SELECT, INSERT, QUICK };
+
+    /** The random number generator. */
+    private Random rand;
+
+    /** Create a new SortTiming object with the random number generator initialised with seed. */
+    public SortTiming(long seed) {
+        rand = new Random(seed);
+    }
+
+    /** Reset the random number generator with the given random seed. */
+    public void setSeed(long seed) { rand.setSeed(seed); }
+
+    /** Generates an array of size elements randomly initialised between lo and hi, inclusive. */
+    private int[] randomlyInitArray(final int size, final int lo, final int hi) {
+        final int range = hi - lo + 1;
+        int[] a = new int[size];
+
+        for (int i = 0; i < size; i++) {
+            a[i] = rand.nextInt(range) + lo;
+        }
+
+        return a;
+    }
+    
+    /** Initialise an array of size elements containing the values [1, 2, ... size]. */
+    private int[] sequentiallyInitArray(final int size) {
+        int[] a = new int[size];
+        
+        for (int i = 0; i < size; i++) {
+            a[i] = i + 1;
+        }
+        
+        return a;
+    }
+
+    /** Compares all supported sorting algorithms on randomised data in range [lo, hi]. */
+    public void compareSortsOnRandom(final int size, final int lo, final int hi, boolean displayAfter) {
+        compareSorts( randomlyInitArray(size, lo, hi), displayAfter );
+    }
+    
+    /** Compares all supported sorting algorithms on sequential data. */
+    public void compareSortsOnSorted(final int size, boolean displayAfter) {
+        compareSorts( sequentiallyInitArray(size), displayAfter );
+    }
+
+    /**
+     * Compares all supported sorting algorithms on the given array of integers, optionally
+     * displaying their results after to confirm they all achieve the same outcome.
+     */
+    public void compareSorts(int[] data, boolean displayAfter) {
+        long bubbleTime, selectTime, insertTime, quickTime;
+        int[] bubbleSorted = Arrays.copyOf(data, data.length);
+        int[] selectSorted = Arrays.copyOf(data, data.length);
+        int[] insertSorted = Arrays.copyOf(data, data.length);
+        int[] quickSorted = Arrays.copyOf(data, data.length);
+
+        ArrayRoutines.setTracing(false);
+
+        System.out.print("\nSorting " + data.length + " items to compare sorting algorithm performance...");
+        bubbleTime = timeSort(bubbleSorted, Alg.BUBBLE);
+        selectTime = timeSort(selectSorted, Alg.SELECT);
+        insertTime = timeSort(insertSorted, Alg.INSERT);
+        quickTime = timeSort(quickSorted, Alg.QUICK);
+        System.out.println("done");
+
+        if (displayAfter) {
+            System.out.println("Original data");
+            ArrayRoutines.display(data);
+            //Use own display so can present them simultaneously
```

Comment:
```suggestion
           
```

Reply:
This was pre defined template. That's why comments left as it was. You can delete those